### ISSUE TITLE
No logs for action

### ISSUE THREAD

**Issue by i8beef** [actions-on-google/smart-home-nodejs#404]
I've got an action that is permanent testing mode. Do these not get logging services or something? I have not been able to get anything to log, and my implementation is suddenly complaining about SYNC issues, but I have no way of troubleshooting without those logs....

**Comment by Fleker** [comment]
Actions should be getting logs regardless of state.

**Comment by i8beef** [comment]
I figure this probably isn't your area, but do you happen to know where I can get support for no logs showing up? Everything works fine, voice commands, report state, etc., I just get no logs to troubleshoot.

**Comment by Fleker** [comment]
This may be related to the filters you're using in Stackdriver to read the logs.

**Comment by i8beef** [comment]
Well, I tried all the following:

1. Went to simulate, view logs.
2. Tried clearing the filter and then selected Google Assistant Action from the drop down. Only logs I can see with this are from back end stuff from over a month ago, and nothing from my app in particular.

Any other ways to get the filter right?

**Comment by Fleker** [comment]
The "Google Assistant Action" filter should contain any sync errors and other things from that context. If you're hosting your fulfillment on Google's services, then you would want to use a different filter like "Cloud Functions" or "Google App Engine". (You can also concatenate multiple filters together with "OR")

**Comment by i8beef** [comment]
I am not hosting fulfillment on Google services, I have my own servers. My understanding was the stackdriver logs provided the logs for the Google side of this conversation, i.e., I say "sync my devices" and assistant says something went wrong. My side isn't even registering the sync request, meaning I need to see the log of what the Google side thinks happened. Is this not where those would show up? Are you saying this is not true?

**Comment by Fleker** [comment]
If your logging is setup like this, it should work even as a test project.

![image](https://user-images.githubusercontent.com/3291635/60103361-785a0a00-972d-11e9-86ae-9158118a9c13.png)

**Comment by i8beef** [comment]
Yup, that's exactly how I have it setup. I can see a handful of these two errors if I go back in time, but that is ALL I get.

![Capture](https://user-images.githubusercontent.com/1186660/60124735-ea434b00-974f-11e9-97f0-c0d885bd48c6.PNG)

**Comment by i8beef** [comment]
Im still having this issue. All I ever seem to get are those two logs. Existing commands and everything work fine, but a "sync devices" is failing, and I am unable to get any logs to troubleshoot.

**Comment by devunwired** [comment]
Hi @i8beef, a `BACKEND_FAILURE` typically means the Google Assistant was unable to reach your fulfillment or it took too long to respond (this would also be consistent with the "empty JSON" message).

> My side isn't even registering the sync request, meaning I need to see the log of what the Google side thinks happened.

If this is the case 100% of the time, then the Actions console may have the wrong fulfillment URL or testing may no longer be enabled for your project (i.e. you have to go back to the console and click **Start Testing** again). If you only see this happen periodically, then it may be a timeout issue with your response.

**Comment by i8beef** [comment]
No the fullfillment URL is correct. I can execute commands and queries just fine on existing devices. I DID finally get a sync devices to go through... I ended up having to unlink the account, where I noticed that my test app wasn't showing up in my list in the Home app anymore, and then spent several hours trying to figure out how to put the thing back into test mode because none of the documented stuff works. What DID end up working was CHANGING the fullfillment URL to something else, and then changing it back. I'm assuming there is something tied to that action in the Actions Console that drives something vital to a test app?

Note, however, that none of this has done anything for the fact that I still can't see any logs, successful or otherwise.

**Comment by devunwired** [comment]
Thanks for the feedback on the issues you experienced re-enabling testing for your project. We are working to improve this process in the console.

StackDriver doesn't currently log successful interactions, and you will typically only see logs for invalid intent responses like the `BACKEND_FAILURE` you posted (`OPEN_AUTH_FAILURE` during account linking is another example). Rather than being an issue with your configuration, I think StackDriver simply may not be logging all the events you would like to see.

It would be really great if you could file a [feature request](https://issuetracker.google.com/issues/new?component=655104&template=1284148) with the types of log events you would like to see for your action.

**Comment by i8beef** [comment]
Ah, based on @Fleker response, it looked like I was missing a lot of logs... so perhaps I am simply seeing these types of errors when the app falls out of testing mode? What are the rules for that? I thought testing mode was supposed to not expire now, and yet I've seen it happen several times now where I had to reenable it somehow. I used to use gactions to enable it but that doesn't seem to function right anymore either.
